Error.prototype.cause - basic docs (#16706)
mdn/content: files/en-us/web/javascript/reference/global_objects/error/cause/index.md

---
title: Error.prototype.cause
slug: Web/JavaScript/Reference/Global_Objects/Error/cause
tags:
  - JavaScript
  - Property
  - Prototype
browser-compat: javascript.builtins.Error.cause
---
{{JSRef}}

The **`cause`** property indicates the specific original cause of an error.

It is used when catching and re-throwing an error with a more-specific or useful error message in order to still have access to the the original error.

## Value

> Any value can be passed as the cause, but an Error is expected.

Reviewer: Beware, an Error would not be expected, and the original proposal is designed with non-caught-error causes in mind. For example:

```js
if (n < 0) {
  throw new RangeError("n should not be negative", { cause: { n } });
}
```

Not to say it happens a lot in practice anyway, given the most common pattern should still be `{ cause: err }`, but worth reconsidering the phrasing here.

Author: Good point. How about.
```suggestion
Any value may be passed as the cause.
```

Once this is in, I suggest you add another example "Throwing something other than an error". Would that be OK? Or you can patch this now.

Reviewer: Yep! I do plan to elaborate on this later. I don't have write access so if I'm going to patch this now it will be in the form of a bunch of suggestion comments which I'd rather not to, since I probably have to play with the words a bit.

This proposed change looks good to me, although we can also mention that it is just the value used in the constructor's `options.cause`.

---
title: Error.prototype.cause
slug: Web/JavaScript/Reference/Global_Objects/Error/cause
tags:
  - JavaScript
  - Property
  - Prototype
browser-compat: javascript.builtins.Error.cause
---
{{JSRef}}

The **`cause`** property indicates the specific original cause of an error.

It is used when catching and re-throwing an error with a more-specific or useful error message in order to still have access to the the original error.

## Value

This is the value that was passed to the [`Error()` constructor](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error) in the `options.cause` argument.

> Any value may used.

Reviewer 2: nitpick
```suggestion
The value can be of any type.
```

Author: Thanks @OnkarRuikar - accepting both suggestions.
